google/cloud/storage/well_known_headers.h

```diff
@@ -62,6 +66,82 @@ struct IfNoneMatchEtag : public WellKnownHeader<IfNoneMatchEtag, std::string> {
   static char const* header_name() { return "If-None-Match"; }
 };
 
+struct EncryptionKeyData {
+  std::string algorithm;
+  std::string key;
+  std::string sha256;
+};
+
+// Compute the encryption data given a (binary) AES256 key.
+EncryptionKeyData EncryptionDataFromBinaryKey(std::string const& key);
+
+struct EncryptionKey
+    : public WellKnownHeader<EncryptionKey, EncryptionKeyData> {
+  using WellKnownHeader<EncryptionKey, EncryptionKeyData>::WellKnownHeader;
+
+  /**
+   * Create an encryption key parameter from a binary key.
+   *
+   * @param key a binary key, must have exactly 8 bytes.
+   */
+  static EncryptionKey FromBinaryKey(std::string const& key);
+
+  static char const* prefix() { return "X-Goog-Encryption-"; }
+};
+
+std::ostream& operator<<(std::ostream& os, EncryptionKey const& rhs);
+
+struct SourceEncryptionKey
+    : public WellKnownHeader<SourceEncryptionKey, EncryptionKeyData> {
+  using WellKnownHeader<SourceEncryptionKey,
+                        EncryptionKeyData>::WellKnownHeader;
+
+  /**
+   * Create a source encryption key parameter from a binary key.
+   *
+   * @param key a binary key, must have exactly 8 bytes.
+   */
+  static SourceEncryptionKey FromBinaryKey(std::string const& key);
+
+  static char const* prefix() { return "X-Copy-Source-Goog-Encryption-"; }
```

Comment:
Nit: Everywhere I see in our docs, these headers are given in all lowercase.  I'd prefer the case we use be consistent with the docs.

Reply:
Done.